internal/extension/smartagentextension/extension.go

```diff
@@ -17,15 +17,28 @@ package smartagentextension
 import (
 	"context"
 
+	saconfig "github.com/signalfx/signalfx-agent/pkg/core/config"
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/config"
 )
 
+// SmartAgentConfigProvider exposes global saconfig.Config to other components
+type SmartAgentConfigProvider interface {
```

Comment:
The config is no longer available via the GetExtensions. You need to expose things on the instance.

Reply:
No longer in this PR.